module/os/linux/zfs/zfs_ctldir.c

```diff
@@ -767,9 +767,6 @@ zfsctl_snapshot_path_objset(zfsvfs_t *zfsvfs, uint64_t objsetid,
 	uint64_t id, pos = 0;
 	int error = 0;
 
-	if (zfsvfs->z_vfs->vfs_mntpoint == NULL)
```

Comment:
Though I am not very deep into this, I am curious why we do not return early like before in the case of `zfsvfs->z_vfs->vfs_mntpoint == NULL`.

Reply:
Between acquiring the lock above and then below again, it seemed simpler and cleaner to acquire once and set the error, if `vfs_mntpoint` is `NULL`. There will be very minor paths left where `vfs_mntpoint` could be `NULL` now.